backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Fresh_PriceList_Details_ToDate_Report.sql

```diff
@@ -0,0 +1,121 @@
+                                                                     text,
+                                                                     date)
+;
+
+CREATE OR REPLACE FUNCTION report.Fresh_PriceList_Details_ToDate_Report(
+    p_ad_language                character varying,
+    p_show_product_price_pi_flag text,
+    p_validfrom                  date
+)
+    RETURNS TABLE
+            (
+                bp_value                   text,
+                bp_name                    text,
+                productcategory            text,
+                m_product_id               integer,
+                value                      text,
+                customerproductnumber      text,
+                productname                text,
+                isseasonfixedprice         character,
+                itemproductname            character varying,
+                qtycuspertu                numeric,
+                packingmaterialname        text,
+                pricestd                   numeric,
+                pricepattern1              text,
+                altpricestd                numeric,
+                pricepattern2              text,
+                hasaltprice                integer,
+                uomsymbol                  text,
+                uom_x12de355               text,
+                attributes                 text,
+                m_productprice_id          integer,
+                m_attributesetinstance_id  integer,
+                m_hu_pi_item_product_id    integer,
+                currency                   character,
+                currency2                  character,
+                show_product_price_pi_flag text,
+                plv_name                   text
+            )
+AS
+$$
+BEGIN
+
+    DROP TABLE IF EXISTS temp_price_list_details;
+
+    CREATE TEMP TABLE IF NOT EXISTS temp_price_list_details AS
+    SELECT DISTINCT bp.c_bpartner_id,
+                    plv.M_PriceList_Version_ID,
+                    plv.name
+    FROM c_bpartner bp
+             INNER JOIN M_PricingSystem ps ON bp.m_pricingsystem_id = ps.m_pricingsystem_id
+             INNER JOIN C_BPartner_Product bpp ON bp.c_bpartner_id = bpp.c_bpartner_id AND bp.iscustomer = 'Y'
+             INNER JOIN M_ProductPrice pp ON pp.M_Product_ID = bpp.M_Product_ID
+             INNER JOIN M_PriceList_Version plv ON plv.M_PriceList_Version_ID = pp.M_PriceList_Version_ID
+             INNER JOIN M_PriceList pl ON plv.m_pricelist_id = pl.m_pricelist_id AND ps.m_pricingsystem_id = pl.m_pricingsystem_id AND pl.issopricelist = 'Y'
+        AND plv.validfrom >= p_validfrom;
+
+
+    IF p_show_product_price_pi_flag = 'N' THEN
+        RETURN QUERY (SELECT DISTINCT d.*, tmp.name::text AS plv_name
+                      FROM temp_price_list_details tmp
+                               INNER JOIN report.fresh_PriceList_Details_Report(
+                              tmp.c_bpartner_id,
```

Comment:
I think the IF is not ok.
In The original report we have an UNION.
I think here shall be the same.

Reply:
- as talked the union is just for the jasper,
 - jasper files removed